### PR TITLE
[fix][metrics] Reduce policy_entropy and logprob-diff metrics via sum/count

### LINKED ISSUE #1822: Fix reduction for entropy and other metrics in policy workers
# Summary

Currently metrics like `entropy` as well as `minibatch_rollout_logprobs_abs_diff_mean ` are aggregated as follows:
1. Masked mean within a micro batch. For example, for entropy: https://github.com/erictang000/SkyRL/blob/4d8da74ca1e8a0b3ffd4bc3844a841c97991a57a/skyrl/backends/skyrl_train/workers/worker.py#L943
2. Reduced across micro batches: https://github.com/erictang000/SkyRL/blob/4d8da74ca1e8a0b3ffd4bc3844a841c97991a57a/skyrl/backends/skyrl_train/workers/worker.py#L1077
3. All-reduced across DP workers: 

https://github.com/erictang000/SkyRL/blob/4d8da74ca1e8a0b3ffd4bc3844a841c97991a57a/skyrl/backends/skyrl_train/workers/worker.py#L1079

However, this can lead to somewhat incorrect metrics compared to computing the same metric at the global batch level. For example, there can be cases where some DP workers only get failed trajectories or get heavily imbalanced nnz values in `loss_mask`. In these cases, the values from some DP workers can contribute more to the final statistics.

I would imagine the right thing to do is to treat the metrics as separate sum loss metrics:

(1) Entropy / rollout logprobs diff summed over all tokens with non-zero loss mask
(2) the loss mask nnz

We should take a sum of these metrics across all DP workers, and then compute the desired metric with a final division.

This will also give us an exact reduction as if the metric were computed with the global mini batch, and alleviate issues like here: https://github.com/NovaSky-AI/SkyRL/pull/1813#issuecomment-4754829694

Original context: https://github.com/NovaSky-AI/SkyRL/pull/1813#discussion_r3449772146

### PR DESCRIPTION
**Pull request by ayucpp**
Fixes #1822

## What does this PR do?

Updates metric aggregation for `policy_entropy` and `minibatch_rollout_logprobs_abs_diff_*` to compute the exact global mean instead of a mean-of-means.

## Why

Workers reported per-micro-batch masked means, averaged across micro-batches, DP ranks, and mini-batches — over-weighting shards with few unmasked tokens.

Example: Shard A = 1 token (entropy 1.0), Shard B = 100 tokens (entropy 2.0):
- Old (mean-of-means): (1.0 + 2.0) / 2 = 1.5
- New (exact): 201 / 101 ≈ 1.99

Token-based batching (or DP workers getting mostly failed trajectories) makes shards imbalanced, so the old value wasn't the global-batch metric.

## What changes

- Workers emit masked per-token sums (`policy_entropy_sum`, `minibatch_rollout_logprobs_abs_diff_sum`/`_sq_sum`) and counts (`loss_mask_nnz`, `..._nnz`); `reduce_metrics`/`all_reduce_metrics` always sum these.
- Final derivations (entropy = sum/count, std = sqrt(E[x²] − E[x]²)) moved to the trainer, computed once.
- Workers still emit a final `policy_entropy` after the DP all-reduce, so Tinker `_extract_metrics` and worker-level GPU assertions are unchanged.
- Megatron packed-entropy helpers now return the CP-global sum/count, keeping CP invariance.

## Testing

- Added imbalanced-shard exactness tests in `test_worker_utils.py` and `test_skip_fwd_logprobs.py`.
- CPU suite: 1393 passed (`-m "not vllm"`).
- Pre-commit clean.

### CODE REVIEW

File: skyrl/backends/skyrl_train/workers/megatron/megatron_model_wrapper.py [hunk 1]
```diff
@@ -929,7 +936,8 @@ def loss_func(logits, data):
             metrics = {
                 "final_loss": unscaled_loss.detach().item(),
                 "policy_loss": policy_loss.detach().item(),
-                "policy_entropy": entropy.detach().item(),
+                POLICY_ENTROPY_SUM_KEY: entropy_sum,
+                LOSS_MASK_NNZ_KEY: entropy_nnz,
```
[the remark this answers is not in the document]
Comment (ayucpp) [reply skyrl/backends/skyrl_train/workers/megatron/megatron_model_wrapper.py:946]: Fixed in ac1318c0 — I normalized to Python floats right after the branch chain (`entropy_sum = float(entropy_sum); entropy_nnz = float(entropy_nnz)`), which covers both the packed branches (tensors) and `compute_masked_sum_and_count` (already floats) in one place. `float(tensor)` on a scalar tensor is equivalent to `.item()`, so the metrics now survive the `isinstance(x, (int, float))` check in `reduce_metrics`. Full CPU suite: 1393 passed, no regressions.
